PicPi/AddPhotoViewController.swift

```diff
@@ -0,0 +1,196 @@
+    )
+    
+  }
+  private func setupTableView() {
+    tableView.allowsMultipleSelection = false
+    tableView.dataSource = self
+    tableView.delegate = self
+    view.addSubview(tableView)
+    tableView.register(UITableViewCell.self, forCellReuseIdentifier: "cell")
+    tableView.translatesAutoresizingMaskIntoConstraints = false
+    NSLayoutConstraint.activate(
+      [
+        tableView.topAnchor.constraint(equalTo: doneButton.bottomAnchor, constant: 10),
+        tableView.leftAnchor.constraint(equalTo: view.leftAnchor),
+        tableView.rightAnchor.constraint(equalTo: view.rightAnchor),
+        tableView.bottomAnchor.constraint(equalTo: sendButton.topAnchor),
+      ]
+    )
+  }
+  
+  // MARK: - Buttons Actions
+  @objc func sendButtonPressed(sender: UIButton!) {
+    photoUploadManager.startUploadPreviouslySelectedPhotos()
+    sendButton.isEnabled = false
+    
+  }
+  @objc func doneButtonPressed(sender: UIButton!) {
+    self.dismiss(animated: true, completion: nil)
+  }
+  
+}
+// MARK: - BSImagePicker Setup
+extension AddPhotoViewController {
+  
+  func showImagePickerController(){
+    
+    let imagePicker = ImagePickerController()
+    imagePicker.modalPresentationStyle = .fullScreen
+    presentImagePicker(imagePicker, select: { (asset) in
+      
+    }, deselect: { (asset) in
+      
+    }, cancel: { (assets) in
+      
+    }, finish: { [self] (assets) in
+      /// User finished picking images, inform upload manger and pass an array of assets to prepear for upload
+      
+      self.photoUploadManager.finishedPickingPhotos(photoAssetes: assets)
+      tableView.reloadData()
+    })
+    
+  }
+  
+}
+
+extension AddPhotoViewController: UITableViewDelegate , UITableViewDataSource {
+  // MARK: - UITableViewDataSource
+  
+  func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
+    let cell = tableView.dequeueReusableCell(withIdentifier: "cell", for: indexPath)
+    if indexPath.section == 0 {
+      let photoCount = photoUploadManager.picturesCount
+      switch photoCount {
+        case 1:
+          cell.textLabel?.text = "\(photoCount) photo selected!!"
+        case let x where x > 1:
+          cell.textLabel?.text = "\(photoCount) photo selected!!"
+          
+        default:
+          cell.textLabel?.text = "No photo selected!!"
+      }
+      
+      
+    }else {
+      cell.textLabel?.text = "PictureFrame 1"
+    }
+    cell.textLabel?.textAlignment = .center
+    return cell
+  }
+  
+  func tableView(_ tableView: UITableView, titleForHeaderInSection section: Int) -> String? {
+    if section == 0 {
+      return "Add Photos"
+    }
+    return  "Send Photos to..."
+  }
+  
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    if section == 0 {
+      return 1
+    }
+    /// return 1 for now , after we start save multiple picture frames IPs we could return the count of IPs
```

Comment:
nit: update this to a TODO styled comment:
```suggestion
    // TODO: after we start save multiple picture frames IPs we could return the count of IPs
```